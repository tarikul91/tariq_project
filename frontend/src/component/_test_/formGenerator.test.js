import { render, screen, cleanup } from "@testing-library/react";
import FormGenerator from "component/FormGenerate"
import renderer from "react-test-renderer"
afterEach(cleanup)

test('FormGenerate without any props', () => {
    render(<FormGenerator></FormGenerator>)
    const container = screen.getByTestId('form-container')
    const form = screen.getByTestId('form')
    const heading = screen.getByTestId('form-heading')
    expect(container).toContainElement(form)
    expect(form).not.toHaveClass()
    expect(heading).not.toHaveTextContent()
})

test('FormGenerate with className formName and style', () => {
    render(<FormGenerator
        className="form"
        formName="Login form"
        style={{ marginBottom: "5px" }}
    ></FormGenerator>)
    const container = screen.getByTestId('form-container')
    const form = screen.getByTestId('form')
    const heading = screen.getByTestId('form-heading')
    expect(container).toContainElement(form)
    expect(form).toHaveClass('form')
    expect(form).toHaveStyle('margin-bottom:5px')
    expect(heading).toHaveTextContent("Login form")
})

test('FormGenerate with children', async () => {
    render(<FormGenerator>
        <button>Submit</button>
    </FormGenerator>)
    const child = screen.getByRole('button', { name: /Submit/i })
    const form = screen.getByTestId('form')
    expect(form).toContainElement(child)
})

it('FormGenerate snapshot', () => {
    const tree = renderer.create(
        <FormGenerator
            className="form"
            formName="Login form"
            style={{ marginBottom: "5px" }}
        >
            <button>Submit</button>
        </FormGenerator>
    ).toJSON()
    expect(tree).toMatchSnapshot()

})