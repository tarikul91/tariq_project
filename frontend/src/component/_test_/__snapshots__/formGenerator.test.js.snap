// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormGenerate snapshot 1`] = `
<div
  className="form-container"
  data-testid="form-container"
>
  <form
    className="form"
    data-testid="form"
    style={
      Object {
        "marginBottom": "5px",
      }
    }
  >
    <h3
      className="text-center m-3"
      data-testid="form-heading"
    >
      Login form
    </h3>
    <button>
      Submit
    </button>
  </form>
</div>
`;
